### ISSUE TITLE
Email from Claudio

### ISSUE THREAD

**Issue by Joniff** [Joniff/Terratype#12]

**Comment by Joniff** [comment]
In regard to item 1, that is a bug, should have read _icon throughout. Have fixed for next release.

In regard to item 2, the proposed solution is for when the map is clicked, not the marker. Currently there is no event raised when you click the map. So this isn't supported.

Have just retested, clicking markers, but not the map, in Gmaps seems to be raising relevant js event.

**Comment by Joniff** [comment]
Have released version 1.0.16 with a fix for Item 1. Will take a few hours for nuget to update itself.

**Comment by claudioacanfora** [comment]
Hi joniff! Thanks very mush for fixing the bug so fast!!

Regarding the second "bug": I was speaking about the click on the marker. Adding my code, even if you don't populate the pop-window, the terratype will trigger the event "onClick". It works 100%,, have a try!

In short, if you add something like this into the view:
`
...
@Html.Terratype(mapOptions, location)
...
terratype.onClick(function (provider, map, marker) {
  console.log("marker clicked!");
}
`

Thanks again,
Claudio

**Comment by Joniff** [comment]
You are correct, it only raises the event if you have a popup label. This should not be the case, I need to amend the code to raise the event regardless of whether a popup label is present or not.

I will get a fix for this out too, stay tuned.
And thanks again for reporting this, great work.

**Comment by PezCo** [comment]
Did fix for point 2 ever get added? currently if i click a marker without any content in the popup the click event doesn't fire.

The reason i want this is so that i can click a marker and open up content that is not google map specific. e.g a modal

**Comment by Joniff** [comment]
This has been fixed in version 1.0.18, Now markers will fire onClick event even if they don't have a label present
